### PR TITLE
Цветков М.С. 381803-2. ЛР2

### PR DESCRIPTION
**Pull request by TsvetkovMaximPmi3**
### Pull Request Readiness Checklist

See details at [code review]( https://github.com/UNN-ITMM-Software/devtools-course-practice/blob/main/lab-guide/lab-1-setup.md#%D1%80%D0%B5%D1%86%D0%B5%D0%BD%D0%B7%D0%B8%D1%8F-%D0%BA%D0%BE%D0%B4%D0%B0)

- [ ] Code is well documented and more information can be seen on the [website](https://devtools-course-practice-2021.readthedocs.io/en/latest/modules.html) (recommended)
- [ ] Make sure that the testing has passed successfully on [GitHub Actions](https://github.com/UNN-ITMM-Software/devtools-course-practice/actions)
- [ ] Get two approve from your fellow students
- [ ] Take a review from one of the administrators

### CODE REVIEW

**Review by alxndrvna — APPROVED** [review]
Хороший код!

**Review by dmitrybezro — APPROVED** [review]
Все хорошо

**Comment by TsvetkovMaximPmi3** [comment]
@dkurt check my request

File: modules/subseq_finder/include/subseq_finder.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#include <algorithm>
+#include <string>
+
+class SubseqFinder {
+ private:
+    std::vector <int> seq_1;
+    std::vector <int> seq_2;
+ public:
+    SubseqFinder(std::vector <int> seq_1,
+                 std::vector <int> seq_2);
```
Comment (dkurt) [inline modules/subseq_finder/include/subseq_finder.h:17]: Передаёте по значению => лишние копирование данных. Сделаейте по константной ссылке

File: modules/subseq_finder/src/subseq_finder.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
+// Copyright 2021 Tsvetkov Maxim
+
+#include <algorithm>
+#include <vector>
+#include <iostream>
+#include "include/subseq_finder.h"
+
+SubseqFinder::SubseqFinder(std::vector <int> seq_1,
+std::vector <int> seq_2):seq_1(seq_1), seq_2(seq_2) {}
```
Comment (dkurt) [inline modules/subseq_finder/src/subseq_finder.cpp:9]: fix alignment

File: modules/subseq_finder/src/subseq_finder.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+std::vector<int> SubseqFinder::findMaxSubseq(std::vector<int> a,
+                                             std::vector<int> b) {
+    int firstSize = a.size();
+    int secondSize = b.size();
+    std::vector<int>result;
+    std::vector<std::vector <int>> sequencer(firstSize + 1,
+     std::vector<int>(secondSize + 1));
```
Comment (dkurt) [inline modules/subseq_finder/src/subseq_finder.cpp:17]: fix code style

File: modules/subseq_finder/include/subseq_finder.h [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    std::vector<int> getMaxSubseq();
+    static std::vector<int> findMaxSubseq(std::vector <int> seq_1,
+                                       std::vector <int> seq_2);
```
Comment (dkurt) [inline modules/subseq_finder/include/subseq_finder.h:20]: Please return result by arguments

Comment (TsvetkovMaximPmi3) [reply modules/subseq_finder/include/subseq_finder.h:20]: I don't understand what do you mean.

File: modules/subseq_finder/.vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,51 @@
+{
```
Comment (dkurt) [inline modules/subseq_finder/.vscode/settings.json:1]: please remove extra file

File: modules/subseq_finder/include/subseq_finder.h (hunk 4 above)
Comment (dkurt) [reply modules/subseq_finder/include/subseq_finder.h]: return of `std::vetor<int>` through the `return` will lead to data copy. Return it by argument:
```cpp
static void findMaxSubseq(const std::vector <int>& seq_1,
                          const std::vector <int>& seq_2,
                          std::vector<int>& res);
```

Comment (TsvetkovMaximPmi3) [reply modules/subseq_finder/include/subseq_finder.h]: Is this a non-const reference? If so, make const or use a pointer: std::vector<int>& result
This is linter output for your suggestion. I think that using pointer here is not a good decision.

Comment (TsvetkovMaximPmi3) [reply modules/subseq_finder/include/subseq_finder.h]: I agree with your suggestion, but linter ruins good ideas

Comment (TsvetkovMaximPmi3) [reply modules/subseq_finder/include/subseq_finder.h]: @dkurt Can I resolve this thread?

Comment (dkurt) [reply modules/subseq_finder/include/subseq_finder.h]: No, let's make linter happy and create a pointer for output.

**Pull request merged**